lost/src/main/java/com/mapzen/android/lost/internal/LocationSettingsResultRequest.java

```diff
@@ -0,0 +1,218 @@
+package com.mapzen.android.lost.internal;
+
+import com.mapzen.android.lost.api.LocationRequest;
+import com.mapzen.android.lost.api.LocationSettingsRequest;
+import com.mapzen.android.lost.api.LocationSettingsResult;
+import com.mapzen.android.lost.api.LocationSettingsStates;
+import com.mapzen.android.lost.api.PendingResult;
+import com.mapzen.android.lost.api.ResultCallback;
+import com.mapzen.android.lost.api.Status;
+
+import android.app.PendingIntent;
+import android.bluetooth.BluetoothAdapter;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.location.LocationManager;
+import android.os.AsyncTask;
+import android.os.Handler;
+import android.support.annotation.NonNull;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.mapzen.android.lost.api.Status.CANCELLED;
+import static com.mapzen.android.lost.api.Status.INTERNAL_ERROR;
+import static com.mapzen.android.lost.api.Status.INTERRUPTED;
+import static com.mapzen.android.lost.api.Status.RESOLUTION_REQUIRED;
+import static com.mapzen.android.lost.api.Status.SETTINGS_CHANGE_UNAVAILABLE;
+import static com.mapzen.android.lost.api.Status.SUCCESS;
+import static com.mapzen.android.lost.api.Status.TIMEOUT;
+
+public class LocationSettingsResultRequest extends PendingResult<LocationSettingsResult> {
```

Comment:
Thanks for the thoughtful feedback. Lets address these questions:

1. Upon more thought, we don't get a real benefit from wrapping the `FutureTask` since the operations are short-lived as you say. It was implemented that way with future complication in mind so we definitely could (and probably should, now that you mention it) simplify it and could use a `FutureTask` directly.
2. We could fake the concurrency and implement something similar to what we do in `await()` but just return the `LocationSettingsResult` via the `ResultCallback` instead of having the method return it.
3. This is an interesting question. I think it might be a drastic move though and break too much with the current parity we have across the rest of the library. I think that faking the concurrency should be enough to balance keeping our library simple and users familiar with the api. I'd vote to try out this simplification before eliminating the API.

How does keeping the api and faking the concurrency sound as a path forward?